src/Fields/sum_of_fields.jl

```diff
@@ -0,0 +1,79 @@
+using Base: @propagate_inbounds
+using Oceananigans.Fields: AbstractField, location
+
+import Adapt: adapt_structure
+import Base: getindex, show, summary
+import Oceananigans.BoundaryConditions: fill_halo_regions!
+import Oceananigans.ImmersedBoundaries: mask_immersed_field!, mask_immersed_field_xy!
+
+"""
+    SumOfFields{N, F}
+
+`SumOfFields` objects hold `N` field/fields and return their sum when indexed.
+"""
+struct SumOfFields{N, LX, LY, LZ, G, T, F} <: AbstractField{LX, LY, LZ, G, T, 3}
+    fields :: F
+      grid :: G
+    
+    function SumOfFields{N}(fields...) where N
+        grid = first(fields).grid
+        loc = location(first(fields))
+
+        all(f.grid == grid for f in fields) || 
```

Comment:
```suggestion
        all(f.grid === grid for f in fields) || 
```

I think object equality is appropriate here. And below. (This will be faster because we do not have to perform numeric comparison)